src/main/java/bio/terra/common/DaoUtils.java

```diff
@@ -113,4 +115,17 @@ public static boolean retryQuery(DataAccessException dataAccessException) {
     return ExceptionUtils.hasCause(dataAccessException, RecoverableDataAccessException.class)
         || ExceptionUtils.hasCause(dataAccessException, TransientDataAccessException.class);
   }
+
+  public static PGobject propertiesToPGobject(ObjectMapper objectMapper, Object properties) {
```

Comment:
That's interesting: I'm actually kinda surprised that you had to do this.  Did writing the data as a vanilla JSON string not work?

Reply:
I don't know if what catalog is doing is better or not, but we cast to jsonb inline in the SQL, and don't use a custom mapper. 

https://github.com/DataBiosphere/terra-data-catalog/blob/main/service/src/main/java/bio/terra/catalog/service/dataset/DatasetDao.java#L83

Reply:
Postgres thought it was an hstore data type when I tried to just use a JSON string

Reply:
@pshapiro4broad casting a JSON string worked! Thanks for the suggestion 🙂